### PR TITLE
Introduce JSON options format

### PR DESCRIPTION
**Pull request by PhilipFackler**
See #173

### CODE REVIEW

File: test/config.h.in
Comment (sblondel) [inline test/config.h.in:1]: Should the two bottom lines just be removed?

File: xolotl/options/CMakeLists.txt
Comment (sblondel) [inline xolotl/options/CMakeLists.txt:1]: It looks like one two many blocks

**Review by sblondel — COMMENTED** [review]
Currently the config options are used in most of the other tests, would it be easy to move that to the new JSON format?

File: test/config.h.in
Comment (PhilipFackler) [reply test/config.h.in:1]: Yes, thank you. I'll take care of that.

**Comment by PhilipFackler** [comment]
> Currently the config options are used in most of the other tests, would it be easy to move that to the new JSON format?

It would be easy enough. Let's talk on Monday about how we want to move forward.

**Pull request merged**